### PR TITLE
Use OWNERS-based labeling for kind/api-change

### LINKED ISSUE #9007: Replace path-label munger with appropriate OWNERS files
The only repo that's using it is k/k.  We have support for equivalent functionality in OWNERS files.  I'm not sure all of these are relevant anymore.

```
^docs/proposals			kind/design
^docs/design			kind/design

# examples:
# pkg/api/types.go
# pkg/api/*/types.go
^pkg/api/([^/]+/)?types.go$    kind/api-change
^pkg/api/([^/]+/)?register.go$ kind/new-api

# examples:
# pkg/apis/*/types.go
# pkg/apis/*/*/types.go
^pkg/apis/[^/]+/([^/]+/)?types.go$    kind/api-change
^pkg/apis/[^/]+/([^/]+/)?register.go$ kind/new-api

# docs which are going away with move to separate doc repo
^docs/getting-started-guides	kind/old-docs
^docs/admin			kind/old-docs
^docs/user-guide		kind/old-docs
^docs/devel             kind/old-docs
^docs/design            kind/old-docs
^docs/proposals         kind/old-docs
```

ref: https://github.com/kubernetes/test-infra/pull/8990

### PR DESCRIPTION
**Pull request by spiffxp**
**What this PR does / why we need it**:

This replaces the following path-label munger config, except
we're using kind/api-change for everything instead of two
different kind/ labels

```
^pkg/api/([^/]+/)?types.go$    kind/api-change
^pkg/api/([^/]+/)?register.go$ kind/new-api
^pkg/apis/[^/]+/([^/]+/)?types.go$    kind/api-change
^pkg/apis/[^/]+/([^/]+/)?register.go$ kind/new-api
```

This fixes https://github.com/kubernetes/test-infra/issues/9007

**Special notes for your reviewer**:
This can be merged independently of us turning path-label off of
mungegithub. While both labeling options are active, they won't
clash, just race. 

```release-note
NONE
```

### CODE REVIEW

File: pkg/api/OWNERS [hunk 1]
```diff
@@ -2,3 +2,7 @@ approvers:
 - api-approvers
 reviewers:
 - api-reviewers
+filters:
+  "([^/]+/)?(register|types)\\.go$":
```
Comment (spiffxp) [inline pkg/api/OWNERS]: The original regexes started with `^pkg/api/` but AIUI this takes effect relative to the location of this OWNERS file

**Review by cjwagner — APPROVED** [review]
/lgtm
/hold

We should enable the `verify-owners` plugin across the kubernetes org so that we can be more confident that these kinds of PRs won't break OWNERS file parsing.
It is probably stable now.

**Comment by thockin** [comment]
Change LGTM - I have precious little context on the hold, though.

/lgtm
/approve

**Pull request merged**